### PR TITLE
Gocloud Architecture

### PR DESCRIPTION
**Pull request by PratikDhanave**
gocloud architecture with delegation pattern
GCE
 Createnode()
 Startnode()
 Stopnode()
 Deletenode()
Rebootnode()
EC2
 Createnode()
 Startnode()
 Stopnode()
 Deletenode()
 Rebootnode()

### CODE REVIEW

**Comment by rehrumesh** [comment]
@PratikDhanave please fix the issues mentioned in codacy to improve code quality.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,17 @@
 # gocloud-v2
 Cloud services library
+
+
+gocloud-v2 is a golang library which hides differences between different cloud providers(AWS,GCP etc) APIs and allows you to manage different cloud resources through a unified and easy to use API.
```
Comment (rehrumesh) [inline README.md]: Rename it to gocloud, this will be merged to mail repo soon after the project

File: aws/aws.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package aws
+
+import(
+      .	"github.com/scorelab/gocloud-v2/compute/ec2"
```
Comment (rehrumesh) [inline aws/aws.go]: Avoid using dot imports

File: compute/ec2/instance.go [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+
+	"encoding/base64"
+	"encoding/xml"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"strconv"
+	"time"
+	//"reflect"
```
Comment (rehrumesh) [inline compute/ec2/instance.go]: Do not include commented code to the PR

**Comment by rehrumesh** [comment]
Also use `gofmt` before submit PR

File: compute/ec2/instance.go [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+	}
+	return resp, nil
+}
+
+
+// delete ec2 instance accept array of instance-id
+
+func (ec2 *EC2) Deletenode(request interface{}) (resp interface{}, err error) {
+	  instIds := request.([]string)
+   	params := makeParams("TerminateInstances")
```
Comment (rehrumesh) [inline compute/ec2/instance.go]: inconstant spacing

File: compute/ec2/instance.go [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+		}
+		return
+}
+
+
+
+//pass the param to query and add signature to it base on secret key and acces key
+
+func (ec2 *EC2) query(params map[string]string, resp interface{}) error {
+
```
Comment (rehrumesh) [inline compute/ec2/instance.go:73]: inconstant line break

File: compute/ec2/instance.go [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+	// Add the params passed in to the query string
+	query := req.URL.Query()
+	for varName, varVal := range params {
+		query.Add(varName, varVal)
+	}
+	query.Add("Timestamp", timeNow().In(time.UTC).Format(time.RFC3339))
+	req.URL.RawQuery = query.Encode()
+
+	auth := Auth{"dummy","dummy"}
```
Comment (rehrumesh) [inline compute/ec2/instance.go]: Why `dummy` is hard coded?

File: compute/ec2/instance.go [hunk 7]
```diff
@@ -0,0 +1,369 @@
...
+	return xml.NewDecoder(r.Body).Decode(resp)
+
+}
+
+
+//create Ec2 instances accept map[string]interface{} with attribute Define in EC2 documentation
+
+func (ec2 *EC2) Createnode(request interface{})(resp interface{}, err error){
+
+		var options RunInstances
```
Comment (rehrumesh) [inline compute/ec2/instance.go]: Was this code copied from some other place?

Comment (rehrumesh) [reply compute/ec2/instance.go]: Code quality is really low in following section.

File: main.go [hunk 8]
```diff
@@ -0,0 +1,195 @@
+package main
+
+import(
+  . "github.com/scorelab/gocloud-v2/gocloud"
+)
+
+
+func main(){
+
+    /*
```
Comment (rehrumesh) [inline main.go]: Why this entire block is commented? If it is not relevant, remove it from the PR.

**Comment by rehrumesh** [comment]
@PratikDhanave why did you close the PR?

**Comment by PratikDhanave** [comment]
@rumesh I created new one.

On 18-Jun-2017 6:27 PM, "Rumesh Eranga Hapuarachchi" <
notifications@github.com> wrote:

@PratikDhanave <https://github.com/pratikdhanave> why did you close the PR?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/scorelab/gocloud-v2/pull/1#issuecomment-309276215>, or mute
the thread

**Pull request closed without merging**